library/ziti.c

```diff
@@ -381,6 +381,11 @@ int ziti_close(ziti_connection *conn) {
 }
 
 int ziti_write(ziti_connection conn, uint8_t *data, size_t length, ziti_write_cb write_cb, void *write_ctx) {
+    if (conn->state != Connected) {
+        ZITI_LOG(ERROR, "attempted write on conn[%d] in invalid state[%d]", conn->conn_id, conn->state);
```

Comment:
Shouldn't this check for `CloseWrite` too? I'd like to see a warning in the logs if i'm writing to a closed connection and maybe return a different value than 'invalid'?

Reply:
not sure what you're getting at -- after you called ziti_close_write() you shouldn't be able to call ziti_write() on same connection

Reply:
ugh. somehow my brain read `!= Connected` as 'disconnected' - i read it numerous times and still brain processed that as 'disconnected'... makes sense now though